spinn_front_end_common/interface/interface_functions/front_end_common_locate_executable_start_types.py

```diff
@@ -0,0 +1,38 @@
+from spinn_front_end_common.utilities import exceptions
+from spinn_utilities.progress_bar import ProgressBar
+
+
+class FrontEndCommonLocateExecutableStartType(object):
+
+    def __call__(
+            self, placements, graph, executable_finder, graph_mapper=None):
+
+        progress = ProgressBar(
+            graph.n_vertices, "Finding executable_start_types")
+        binary_start_type = None
+        for vertex in progress.over(graph.vertices):
+            placement = placements.get_placement_of_vertex(vertex)
+            associated_vertex = graph_mapper.get_application_vertex(vertex)
+            placement_binary_start_type = \
+                associated_vertex.get_binary_start_type()
```

Comment:
the orignal code didnt check either then. lol. i copied it directly from the binary gatherer. lol